app/src/main/java/org/example/viotester/AlgorithmWorker.java

```diff
@@ -339,12 +366,7 @@ public void onSensorChanged(SensorEvent event) {
                     // Event is reused, extract values we want
                     final long time = event.timestamp;
                     final float[] measurement = {event.values[0], event.values[1], event.values[2]};
-                    mNativeHandler.post(new Runnable() {
-                        @Override
-                        public void run() {
-                            processAccSample(time, measurement[0], measurement[1], measurement[2]);
-                        }
-                    });
+                    processAccSample(time, measurement[0], measurement[1], measurement[2]);
```

Comment:
This callback should be already running in the same thread where it would go with `mNativeHandler.post(...)` because the sensors manager is essentially asked to post them to the Handler/event queue of this thread here https://github.com/AaltoML/android-viotester/blob/b465f360a72585b6adc70be86c8a267e840aaa83/app/src/main/java/org/example/viotester/AlgorithmWorker.java#L239
So not in this case. (The situation is different for, e.g., the GPS listener whose callback can come from some other thread). Also note that there are also some added mutexes in the native part of this code that were not there before. This is beacuse some of the (native) operations that touch the same data need to happen in the OpenGL thread